libbeat/template/template.go

```diff
@@ -41,6 +45,39 @@ func New(beatVersion string, beatName string, esVersion string, config TemplateC
 		pattern = name + "-*"
 	}
 
+	compiler := func(key string, ops []fmtstr.VariableOp) (fmtstr.FormatEvaler, error) {
```

Comment:
Any plan to handle `ops`? Event format strings can have defaults if the key is missing in the event. See: https://github.com/elastic/beats/blob/master/libbeat/common/fmtstr/formatevents.go#L280

One can use `%{[my.field]:defaultValue}` in event format strings.